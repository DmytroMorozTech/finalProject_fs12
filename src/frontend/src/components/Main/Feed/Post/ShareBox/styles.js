import {makeStyles} from '@material-ui/core/styles'
export default makeStyles(() => ({
  share: {
    display: 'flex',
    flexDirection: 'column',
    cursor: 'pointer',
    border: '1px solid rgba(0, 0, 0, 0.5)',
    borderRadius: '10px',
    width: '45%'
  },
  post: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    marginLeft: '8px',
    color: 'gray',
    '& > .MuiSvgIcon-root': {
      fontSize: 40
    }
  },
  postButton: {
    margin: '4px 4px',
    flexGrow: '1',
    paddingLeft: '16px',
    border: '1px solid rgba(0, 0, 0, 0.15)',
    borderRadius: '35px',
    textAlign: 'left',
    height: '30px',
    backgroundColor: 'white',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  shareButtons: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  photo: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#0a66c2',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  video: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#3ACB0A',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  event: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#F8A558',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  article: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#E6699D',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  names: {
    color: 'gray',
    fontSize: 12,
    fontWeight: 400,
    marginLeft: '10px'
  }
}))