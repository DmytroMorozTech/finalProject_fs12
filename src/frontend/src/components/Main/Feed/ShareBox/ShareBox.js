import React, {useState} from 'react'
import Style from './styles'
import PhotoSizeSelectActualIcon from '@material-ui/icons/PhotoSizeSelectActual'
import YouTubeIcon from '@material-ui/icons/YouTube'
import EventNoteIcon from '@material-ui/icons/EventNote'
import AssignmentIcon from '@material-ui/icons/Assignment'
import Avatar from '../../../../shared/Avatar/Avatar'

function ShareBox (props) {
  const classes = Style()
  const [showModal, setShowModal] = useState('close')
  const clickHandler = (event) => {
    event.preventDefault()
    if (event.target !== event.currentTarget) {
      return
    }
    switch (showModal) {
      case 'open':
        setShowModal('close')
        break
      case 'close':
        setShowModal('open')
        break
      default:
        setShowModal('close')
        break
    }
  }
  return (
    <div className={classes.share}>
      <div className={classes.post}>
        <Avatar/>
        <button className={classes.postButton} onClick={clickHandler} disabled={!!props.loading}>
                        New post
        </button>
      </div>
      <div className={classes.shareButtons}>
        <div className={classes.photo}>
          <PhotoSizeSelectActualIcon/>
          <div className={classes.names}>Photo</div>
        </div>
        <div className={classes.video}>
          <YouTubeIcon/>
          <div className={classes.names}>Video</div>
        </div>
        <div className={classes.event}>
          <EventNoteIcon/>
          <div className={classes.names}>Event</div>
        </div>
        <div className={classes.article}>
          <AssignmentIcon/>
          <div className={classes.names}>Write Article</div>
        </div>
      </div>
    </div>
  )
}
export default ShareBox