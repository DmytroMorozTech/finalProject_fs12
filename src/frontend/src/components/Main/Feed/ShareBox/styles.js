import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({
  share: {
    display: 'flex',
    flexDirection: 'column',
    border: '1px solid #e0e0e0',
    borderRadius: '10px',
    width: '100%'
  },
  post: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    marginRight: '10px',
    marginTop: '4px'
  },
  avatar: {
    marginTop: '5px',
    marginLeft: '20px'
  },
  postButton: {
    margin: '0 10px',
    flexGrow: '1',
    paddingLeft: '16px',
    border: '1px solid #e0e0e0',
    borderRadius: '35px',
    textAlign: 'left',
    height: '50px',
    backgroundColor: 'white',
    cursor: 'pointer',
    color: 'grey',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  shareButtons: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  photo: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#0a66c2',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  video: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#3ACB0A',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  event: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#F8A558',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  article: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    color: '#E6699D',
    transition: 'all 0.35s ease',
    padding: '10px',
    borderRadius: '5px',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.10)'
    }
  },
  names: {
    color: 'grey',
    fontSize: 14,
    fontWeight: 400,
    marginLeft: '10px',
    cursor: 'pointer'
  }
}))