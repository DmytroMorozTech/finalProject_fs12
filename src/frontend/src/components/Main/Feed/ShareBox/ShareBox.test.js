import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import ShareBox from './ShareBox'

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<ShareBox {...props}/>)

describe('should render ShareBox component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let activeUser

  let component
  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    activeUser = {
      'id': 10,
      'createdDate': '2016-07-08T12:22:23.123123',
      'firstName': 'Barry',
      'lastName': 'Evans',
      'fullName': 'Barry Evans',
      'avatarPublicId': 'linkedin/avatars/vnxm3wl3erio9ajtezlr',
      'profileBgPublicId': '',
      'positionAndCompany': 'Scala developer at Amazon.com',
      'email': 'test10@gmail.com',
      'isFollowedByActiveUser': false,
      'numberOfFollowers': 0
    }

    component = setUp({activeUser})
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  // it('should work accept dispatch', () => {
  //   expect(mockDispatch).toHaveBeenCalledTimes(0)
  //   component.find('SharedButton').simulate('click')
  //   expect(mockDispatch).toHaveBeenCalledTimes(1)
  // })
  // it('should work delete dispatch', () => {
  //   expect(mockDispatch).toHaveBeenCalledTimes(0)
  //   component.find('SharedLinkSquare').simulate('click')
  //   expect(mockDispatch).toHaveBeenCalledTimes(1)
  // })
})
