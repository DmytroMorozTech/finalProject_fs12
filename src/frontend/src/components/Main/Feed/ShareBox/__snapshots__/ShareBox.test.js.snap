// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShareBox component to match snapshot 1`] = `
<div>
  <Link
    to="/profiles/10"
  >
    <Image
      alt="user avatar"
      imageUrl="linkedin/avatars/vnxm3wl3erio9ajtezlr"
      type="largeAvatar"
    />
  </Link>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Start a post
  </button>
</div>
`;
